src/datomic_toolbox/core.clj

```diff
@@ -21,97 +20,28 @@
 
 (defn db [] (d/db (connection)))
 
+(defn tempid
+  ([]  (d/tempid (partition)))
+  ([n] (d/tempid (partition) n)))
+
 (defn transact [tx-data]
```

Comment:
I was referring to `transact`. I'm not quite clear on your logic. Are you saying that stuff outside of `core` shouldn't refer to `core/connection`? If that's true, we should remove at least the single-arity version of `swap-tx!`. Or move it in here.